### ISSUE TITLE
Add support to load complex number when `weights_only=True` in `torch.load`

### ISSUE THREAD

**Issue by hzhangxyz** [pytorch/pytorch#141537]
### 🚀 The feature, motivation and pitch

Currently, torch.load does not support builtin complex with `weights_only=True`, for example:

```python
import torch

torch.save(1., "ff")
torch.load("ff", weights_only=True)

torch.save(1.j, "ff")
torch.load("ff", weights_only=True)
```

The result is:
```
Traceback (most recent call last):
  File "/home/hzhangxyz/Cloud/Desktop/qmb/test.py", line 7, in <module>
    torch.load("ff", weights_only=True)
  File "/home/hzhangxyz/Cloud/Desktop/qmb/env/lib/python3.12/site-packages/torch/serialization.py", line 1359, in load
    raise pickle.UnpicklingError(_get_wo_message(str(e))) from None
_pickle.UnpicklingError: Weights only load failed. This file can still be loaded, to do so you have two options, do those steps only if you trust the source of the checkpoint.
        (1) Re-running `torch.load` with `weights_only` set to `False` will likely succeed, but it can result in arbitrary code execution. Do it only if you got the file from a trusted source.
        (2) Alternatively, to load with `weights_only=True` please check the recommended steps in the following error message.
        WeightsUnpickler error: Unsupported global: GLOBAL builtins.complex was not an allowed global by default. Please use `torch.serialization.add_safe_globals([complex])` to allowlist this global if you trust this class/function.

Check the documentation of torch.load to learn more about types accepted by default with weights_only https://pytorch.org/docs/stable/generated/torch.load.html.
```

### Alternatives

I can add complex to safe list as the hint by the error said: `torch.serialization.add_safe_globals([complex])` .

### Additional context

It seems it is because python pickle treat complex type as a non-primitive type. But in reality， complex is primitive indeed. Maybe we could add `"builtins.complex": complex` in `def _get_allowed_globals():` ?

cc @mruberry @mikaylagawarecki @ezyang @anjali411 @dylanbespalko @nikitaved @amjames

**Comment by ezyang** [comment]
Yeah this is a simple oversight. If you can find where to update send us a patch

**Comment by hzhangxyz** [comment]
Sorry, I find this issue has already been fixed (#140850).
